src/cmd/services/m3query/config/config.go

```diff
@@ -101,10 +101,8 @@ var (
 	// users do not experience see unexpected results.
 	defaultRequireExhaustive = true
 
-	defaultWriteWorkerPool = xconfig.WorkerPoolPolicy{
-		GrowOnDemand:          true,
-		Size:                  4096,
-		KillWorkerProbability: 0.001,
+	defaultWorkerPoolPolicy = xconfig.WorkerPoolPolicy{
+		NumShards: 4096,
```

Comment:
i was surprised to learn that both the read and write worker pools were sized to 4096. not sure if that was intentional, but I preserved it.

Reply:
Hm, they should both grow on demand yeah? They should be dynamic worker pools I believe.

Reply:
see the other comment.

Reply:
also for dynamic pools, the size is more than just the initial size. it's the stable size. any goroutines spawned after the stable size will eventually be killed and not respawned. so it's probably still a good idea to size a dynamic pool appropriately to avoid the overhead of respawning goroutines.